src/main/java/seedu/address/model/cheese/CheeseId.java

```diff
@@ -10,18 +10,30 @@
      *
      * @param id A valid id.
      */
-    public CheeseId(int id) {
+    protected CheeseId(int id) {
         super(id);
-        updateNextId(this);
     }
 
     public static CheeseId getNextId() {
-        return new CheeseId(nextId);
+        return getNextId(nextId);
     }
 
+
+    public static CheeseId getNextId(int id) {
+        CheeseId result = new CheeseId(id);
+        updateNextId(result);
+        return result;
+    }
+
+
     private static void updateNextId(CheeseId otherId) {
         if (nextId <= otherId.value) {
             nextId = otherId.value + 1;
         }
     }
+
+    // To be used by stub
```

Comment:
Was thinking that CheeseId should not be doing things for the sake of the stub (i.e. not know of the existence of the stub). Can just specify that it returns the next Id?

Reply:
Okay will specify that it is getter for the fields. But this is still necessary because we want to keep it hidden from the rest of the classes.

Reply:
But note that these changes are essential for us to have the objects be testable. Or else, you can't instantiate a ID with some particular fields without changing the underlying counter. Previously, unlike CustomerId; OrderId and CheeseId are instantiated with native constructor, with no way of not incrementing the underlying counter.

Reply:
By the way, the comment was carried over from seedu.